server/app/com/xsn/explorer/tasks/CurrencySynchronizerTask.scala

```diff
@@ -45,25 +41,16 @@ class CurrencySynchronizerTask @Inject()(
     logger.info("Starting currency synchronizer task")
 
     val currencySynchronizerActor = actorSystem.actorOf(Props[CurrencySynchronizerActor], "currency_synchronizer")
-    actorSystem.scheduler.schedule(config.initialDelay, config.interval) {
-      currencyService.getUSDPrice.onComplete {
-        case Success(Good(usd)) =>
-          logger.info(s"USD price synced to: $usd")
-          currencySynchronizerActor ! CurrencySynchronizerActor.UpdateUSD(usd)
-        case Success(Bad(error)) =>
-          logger.info(s"USD syncronization failed due to $error")
-        case Failure(exception) =>
-          logger.info(s"USD syncronization failed due to $exception")
-      }
 
-      currencyService.getEURPrice.onComplete {
-        case Success(Good(eur)) =>
-          logger.info(s"EUR price synced to: $eur")
-          currencySynchronizerActor ! CurrencySynchronizerActor.UpdateEUR(eur)
+    actorSystem.scheduler.schedule(config.initialDelay, config.interval) {
+      currencyService.getPrices.onComplete {
+        case Success(Good(prices)) =>
+          logger.info("Prices synced")
+          currencySynchronizerActor ! CurrencySynchronizerActor.UpdatePrices(prices)
         case Success(Bad(error)) =>
-          logger.info(s"EUR syncronization failed due to $error")
+          logger.info(s"Prices syncronization failed due to $error")
         case Failure(exception) =>
-          logger.info(s"EUR syncronization failed due to $exception")
+          logger.info(s"Prices syncronization failed due to $exception")
```

Comment:
The right way to log exceptions is to send them as arguments: `logger.info(s"Prices syncronization failed due to ${exception.getMessage}", exception)`.